Bottom: 565cdf686f2b7e5e3794cc1537299e1978cdf8e4
Top:    2333dae67e256c7c8e7dedaa29ed6ac3a1ca7bbf
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-01-07 00:50:22 +0000

Removei SetDefaults parameter from CreateNewPCB and use_default from CreateNewPCBPost

The SetDefaults parameter to CreateNewPCB was not used (despite various callers
passing either true or false).

The use_default parameter to CreateNewPCBPost only had callers passing '1' at this
point, so the parameter is unnecessary.


---

diff --git a/src/action.c b/src/action.c
index b8b3fe7..c886d79 100644
--- a/src/action.c
+++ b/src/action.c
@@ -5938,8 +5938,8 @@ ActionNew (int argc, char **argv, Coord x, Coord y)
 	SaveInTMP ();
       RemovePCB (PCB);
       PCB = NULL;
-      PCB = CreateNewPCB (true);
-      CreateNewPCBPost (PCB, 1);
+      PCB = CreateNewPCB ();
+      CreateNewPCBPost (PCB);
 
       /* setup the new name and reset some values to default */
       free (PCB->Name);
diff --git a/src/buffer.c b/src/buffer.c
index 3b51ba1..789985d 100644
--- a/src/buffer.c
+++ b/src/buffer.c
@@ -1121,7 +1121,7 @@ ConvertBufferToElement (BufferType *Buffer)
 bool
 LoadLayoutToBuffer (BufferType *Buffer, char *Filename)
 {
-  PCBType *newPCB = CreateNewPCB (false);
+  PCBType *newPCB = CreateNewPCB ();
 
   /* new data isn't added to the undo list */
   if (!ParsePCB (newPCB, Filename))
diff --git a/src/create.c b/src/create.c
index 7da96ee..013c8d9 100644
--- a/src/create.c
+++ b/src/create.c
@@ -146,7 +146,7 @@ pcb_colors_from_settings (PCBType *ptr)
  * creates a new PCB
  */
 PCBType *
-CreateNewPCB (bool SetDefaultNames)
+CreateNewPCB (void)
 {
   PCBType *ptr;
   int i;
@@ -220,21 +220,19 @@ CreateNewPCB (bool SetDefaultNames)
  * pre-existing PCB.
  */
 int
-CreateNewPCBPost (PCBType *pcb, int use_defaults)
+CreateNewPCBPost (PCBType *pcb)
 {
   /* copy default settings */
   pcb_colors_from_settings (pcb);
 
-  if (use_defaults)
-    {
-      if (ParseGroupString (Settings.Groups, &pcb->LayerGroups, &pcb->Data->LayerN))
-	return 1;
+  if (ParseGroupString (Settings.Groups, &pcb->LayerGroups, &pcb->Data->LayerN))
+    return 1;
+
+  pcb->Data->Layer[top_silk_layer].Name = strdup ("top silk");
+  pcb->Data->Layer[bottom_silk_layer].Name = strdup ("bottom silk");
+  pcb->Data->Layer[top_soldermask_layer].Name = strdup ("top soldermask");
+  pcb->Data->Layer[bottom_soldermask_layer].Name = strdup ("bottom soldermask");
 
-      pcb->Data->Layer[top_silk_layer].Name = strdup ("top silk");
-      pcb->Data->Layer[bottom_silk_layer].Name = strdup ("bottom silk");
-      pcb->Data->Layer[top_soldermask_layer].Name = strdup ("top soldermask");
-      pcb->Data->Layer[bottom_soldermask_layer].Name = strdup ("bottom soldermask");
-    }
   return 0;
 }
 
diff --git a/src/create.h b/src/create.h
index e9c2697..f352e06 100644
--- a/src/create.h
+++ b/src/create.h
@@ -39,10 +39,10 @@ void CreateBeLenient (bool);
 
 DataType * CreateNewBuffer (void);
 void pcb_colors_from_settings (PCBType *);
-PCBType * CreateNewPCB (bool);
+PCBType * CreateNewPCB ();
 /* Called after PCB->Data->LayerN is set.  Returns zero if no errors,
    else nonzero.  */
-int CreateNewPCBPost (PCBType *, int /* set defaults */);
+int CreateNewPCBPost (PCBType *);
 PinType * CreateNewVia (DataType *, Coord, Coord, Coord, Coord, Coord, Coord, char *, FlagType);
 LineType * CreateDrawnLineOnLayer (LayerType *, Coord, Coord, Coord, Coord, Coord, Coord, FlagType);
 LineType * CreateNewLineOnLayer (LayerType *, Coord, Coord, Coord, Coord, Coord, Coord, FlagType);
diff --git a/src/file.c b/src/file.c
index 3abca69..a8ee601 100644
--- a/src/file.c
+++ b/src/file.c
@@ -386,7 +386,7 @@ real_load_pcb (char *Filename, bool revert)
 {
   const char *unit_suffix, *grid_size;
   char *new_filename;
-  PCBType *newPCB = CreateNewPCB (false);
+  PCBType *newPCB = CreateNewPCB ();
   PCBType *oldPCB;
 #ifdef DEBUG
   double elapsed;
@@ -408,7 +408,7 @@ real_load_pcb (char *Filename, bool revert)
     {
       RemovePCB (oldPCB);
 
-      CreateNewPCBPost (PCB, 1);
+      CreateNewPCBPost (PCB);
       ResetStackAndVisibility ();
 
       /* update cursor location */
diff --git a/src/main.c b/src/main.c
index a217134..673a7b9 100644
--- a/src/main.c
+++ b/src/main.c
@@ -1990,10 +1990,10 @@ main (int argc, char *argv[])
     }
 
   /* Create a new PCB object in memory */
-  PCB = CreateNewPCB (true);
+  PCB = CreateNewPCB ();
   ParseGroupString (Settings.Groups, &PCB->LayerGroups, &PCB->Data->LayerN);
   /* Add silk layers to newly created PCB */
-  CreateNewPCBPost (PCB, 1);
+  CreateNewPCBPost (PCB);
   if (argc > 1)
     command_line_pcb = argv[1];
 
diff --git a/src/parse_y.y b/src/parse_y.y
index fe675a2..eaa92c7 100644
--- a/src/parse_y.y
+++ b/src/parse_y.y
@@ -196,7 +196,7 @@ parsepcb
 			{
 			  PCBType *pcb_save = PCB;
 
-			  CreateNewPCBPost (yyPCB, 1);
+			  CreateNewPCBPost (yyPCB);
 			/* initialize the polygon clipping now since
 			 * we didn't know the layer grouping before.
 			 */
@@ -226,7 +226,7 @@ parsepcb
 		    if (yyPCB != NULL)
 		      {
 			/* This case is when we load a footprint with file->open, or from the command line */
-			CreateNewPCBPost (yyPCB, 1);
+			CreateNewPCBPost (yyPCB);
 			ParseGroupString("1,c:2,s", &yyPCB->LayerGroups, &yyData->LayerN);
 			e = yyPCB->Data->Element->data; /* we know there's only one */
 			PCB = yyPCB;
